multimodal/vl2l/src/mlperf_inference_multimodal_vl2l/task.py

```diff
@@ -230,7 +236,88 @@ async def _query_endpoint_async(
+                    self.event_loop,
+                )
+
+        def _issue_streaming_queries(
+                query_samples: list[lg.QuerySample]) -> None:
+            """Called by the LoadGen to issue queries to the inference endpoint.
+
+            Args:
+                query_samples: The list of query samples to issue to the inference
+                    endpoint. Each query sample contains (1) `id`: `lg.ResponseId`
+                    (i.e., unique identifier for the response), and (2) `index`:
+                    `lg.QuerySampleIndex` (i.e., the sample index into the dataset).
+            """
+
+            async def _query_endpoint_async(
+                    query_sample: lg.QuerySample) -> None:
+                """Query the endpoint through the async OpenAI API client."""
+                messages = self.loaded_messages[query_sample.index]
+                logger.trace(
+                    "Issuing query sample index: {} with response ID: {}",
+                    query_sample.index,
+                    query_sample.id,
+                )
+                ttft_set = False
+                word_array = []
+                stream = await self.openai_api_client.chat.completions.create(
+                    stream=True,
+                    model=self.model_cli.repo_id,
+                    messages=messages,
+                    stream_options={"include_usage": True},
+                )
+                # iterate asynchronously
+                total_tokens = 0
+                async for chunk in stream:
+
+                    # This is the final chunk and will not have 'choices'
+                    if chunk.usage is not None:
+                        total_tokens = int(chunk.usage.completion_tokens)
+
+                    # If it's not the usage chunk, process it as a content
+                    # chunk
+                    choices = getattr(chunk, "choices", None)
+                    if not choices:
+                        continue
+                    # first non-empty token -> TTFT
+                    delta = chunk.choices[0].delta
+                    text = getattr(delta, "content", None)
+                    if text:
+                        if ttft_set is False:
+                            bytes_array = array.array(
+                                "B", text.encode("utf-8"))
+                            address, length = bytes_array.buffer_info()
+                            size_in_bytes = length * bytes_array.itemsize
+                            lg.FirstTokenComplete([
+                                lg.QuerySampleResponse(query_sample.id,
+                                                       address,
+                                                       size_in_bytes,
+                                                       1),
+                            ])
+                            ttft_set = True
+                        word_array.append(text)
```

Comment:
```suggestion
                    if not text:
                        continue
                    if ttft_set is False:
                        bytes_array = array.array(
                            "B", text.encode("utf-8"))
                        address, length = bytes_array.buffer_info()
                        size_in_bytes = length * bytes_array.itemsize
                        lg.FirstTokenComplete([
                            lg.QuerySampleResponse(query_sample.id,
                                                   address,
                                                   size_in_bytes,
                                                   1),
                        ])
                        ttft_set = True
                    word_array.append(text)
```
In general, the less indentention (i.e., nested levels) you have, the more readable the code is. Ruff even has a rule for this: https://docs.astral.sh/ruff/rules/too-many-nested-blocks/